### DISCUSSION TITLE
Basic Struct

### DISCUSSION THREAD

**Discussion by AFFalcons** [rwf2/Rocket#2095] [Questions]
I am trying to write a web server using the v5 release, and know that in v4 there was a Basic Struct (rocket::http::hyper::header::Basic), that was helpful, but it does not exist or was moved in version 5 and was wondering if anyone knew of a replacement or where it was moved?

**Comment by admwrd** [comment]
Since Rocket is importing `hyper`, what you're looking for is how hyper currently handles HTTP Basic Auth.

If you look at [rocket_http](https://github.com/SergioBenitez/Rocket/tree/master/core/http), you can see that its using [hyper version 0.14.9](https://github.com/SergioBenitez/Rocket/blob/master/core/http/Cargo.toml#L50)

In browsing through the hyper repo, I see an issue that brings up your question: https://github.com/hyperium/hyper/issues/1950

That should give you some good information from the source 😃
